### PR TITLE
add watch on csi-block object

### PR DESCRIPTION
**Pull request by bvered**
Signed-off-by: 662962756 <bvered@il.ibm.com>

### CODE REVIEW

**Comment by bvered** [comment]
[Fix CSI driver installation in case it doesn't exists (FSC reconcile is not being triggered)](https://trello.com/c/OhEgiSnH/190-fix-csi-driver-installation-in-case-it-doesnt-exists-fsc-reconcile-is-not-being-triggered)

File: controllers/flashsystemcluster_controller.go [hunk 1]
```diff
@@ -53,6 +58,49 @@ type SecretMapper struct {
...
+	requests := []reconcile.Request{}
+	for _, c := range clusters.Items {
+		req := reconcile.Request{
+			NamespacedName: types.NamespacedName{
+				Namespace: c.GetNamespace(),
+				Name:      c.GetName(),
+			},
+		}
+		requests = append(requests, req)
+		s.reconciler.Log.Info("Discovered a change in CSI CR. Reconciling FlashSystemCluster", "CSIToClusterMapFunc", requests)
```
Comment (tyichye19) [inline controllers/flashsystemcluster_controller.go:98]: consider moving this log outside the loop

File: controllers/util/condition.go [hunk 2]
```diff
@@ -75,6 +74,13 @@ func SetReconcileErrorCondition(conditions *[]odfv1alpha1.Condition, reason stri
 		Reason:  reason,
 		Message: message,
 	})
+
+	SetStatusCondition(conditions, odfv1alpha1.Condition{
```
Comment (tyichye19) [inline controllers/util/condition.go:78]: I saw we call to `SetReconcileErrorCondition` in many cases other than missing CSI CR, so I am not sure we need to change this status here.

File: controllers/flashsystemcluster_controller.go (hunk 1 above)
Comment (bvered) [reply controllers/flashsystemcluster_controller.go:101]: fixed

File: controllers/util/condition.go (hunk 2 above)
Comment (bvered) [reply controllers/util/condition.go:89]: added SetReconcileProvisionerErrorCondition

**Review by tyichye19 — APPROVED** [review]
LGTM

**Pull request merged**